Installation issue: dorado

### Steps to reproduce the issue

```console
gavinjohn@n001:/ref/djslab/software/spack-unstable $ ./bin/spack spec -I dorado
Input spec
--------------------------------
 -   dorado

Concretized
--------------------------------
 -   dorado@0.5.1%gcc@8.5.0~cuda~ipo build_system=cmake build_type=Release generator=make patches=c3f516a arch=linux-rocky8-x86_64
[^]      ^autoconf@2.69%gcc@8.5.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-rocky8-x86_64
[^]          ^m4@1.4.19%gcc@8.5.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rocky8-x86_64
[^]              ^diffutils@3.9%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]              ^libsigsegv@2.14%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]          ^perl@5.36.0%gcc@8.5.0+cpanm+open+shared+threads build_system=generic arch=linux-rocky8-x86_64
[^]              ^berkeley-db@18.1.40%gcc@8.5.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-rocky8-x86_64
[^]              ^gdbm@1.23%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]                  ^readline@8.2%gcc@8.5.0 build_system=autotools patches=bbf97f1 arch=linux-rocky8-x86_64
[^]      ^automake@1.16.5%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]      ^cmake@3.26.3%gcc@8.5.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-rocky8-x86_64
[^]          ^ncurses@6.4%gcc@8.5.0~symlinks+termlib abi=none build_system=autotools arch=linux-rocky8-x86_64
[^]      ^cuda@11.8.0%gcc@8.5.0~allow-unsupported-compilers~dev build_system=generic arch=linux-rocky8-x86_64
[^]          ^libxml2@2.10.3%gcc@8.5.0~python build_system=autotools arch=linux-rocky8-x86_64
[^]      ^curl@8.0.1%gcc@8.5.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2~nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-rocky8-x86_64
[^]          ^pkgconf@1.9.5%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[+]      ^gcc-runtime@8.5.0%gcc@8.5.0 build_system=generic arch=linux-rocky8-x86_64
[^]      ^git@2.40.0%gcc@8.5.0+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=linux-rocky8-x86_64
[^]          ^expat@2.5.0%gcc@8.5.0+libbsd build_system=autotools arch=linux-rocky8-x86_64
[^]              ^libbsd@0.11.7%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]                  ^libmd@1.0.4%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]          ^gettext@0.21.1%gcc@8.5.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-rocky8-x86_64
[^]              ^tar@1.34%gcc@8.5.0 build_system=autotools zip=pigz arch=linux-rocky8-x86_64
[^]                  ^pigz@2.7%gcc@8.5.0 build_system=makefile arch=linux-rocky8-x86_64
[^]          ^libiconv@1.17%gcc@8.5.0 build_system=autotools libs=shared,static arch=linux-rocky8-x86_64
[^]          ^libidn2@2.3.4%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]              ^libunistring@1.1%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]          ^libtool@2.4.7%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]          ^openssh@9.3p1%gcc@8.5.0+gssapi build_system=autotools arch=linux-rocky8-x86_64
[^]              ^krb5@1.20.1%gcc@8.5.0+shared build_system=autotools arch=linux-rocky8-x86_64
[^]                  ^bison@3.8.2%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]              ^libedit@3.1-20210216%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]              ^libxcrypt@4.4.33%gcc@8.5.0~obsolete_api build_system=autotools arch=linux-rocky8-x86_64
[^]          ^pcre2@10.42%gcc@8.5.0~jit+multibyte build_system=autotools arch=linux-rocky8-x86_64
[+]      ^gmake@4.4.1%gcc@8.5.0~guile build_system=generic arch=linux-rocky8-x86_64
[+]      ^hdf5@develop-1.15%gcc@8.5.0+cxx~fortran+hl~ipo~java~map+mpi+shared~subfiling+szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=linux-rocky8-x86_64
[+]          ^libaec@1.0.6%gcc@8.5.0~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-rocky8-x86_64
[^]          ^openmpi@4.1.5%gcc@8.5.0~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~orterunprefix+romio+rsh~singularity+static+vt+wrapper-rpath build_system=autotools fabrics=none schedulers=none arch=linux-rocky8-x86_64
[^]              ^hwloc@2.9.1%gcc@8.5.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rocky8-x86_64
[^]                  ^libpciaccess@0.17%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]                      ^util-macros@1.19.3%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]              ^numactl@2.0.14%gcc@8.5.0 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-rocky8-x86_64
[^]              ^pmix@4.2.3%gcc@8.5.0~docs+pmi_backwards_compatibility~python~restful build_system=autotools arch=linux-rocky8-x86_64
[^]                  ^libevent@2.1.12%gcc@8.5.0+openssl build_system=autotools arch=linux-rocky8-x86_64
[+]      ^htslib@1.15.1%gcc@8.5.0+libcurl+libdeflate build_system=autotools arch=linux-rocky8-x86_64
[^]          ^bzip2@1.0.8%gcc@8.5.0~debug~pic+shared build_system=generic arch=linux-rocky8-x86_64
[^]          ^xz@5.4.1%gcc@8.5.0~pic build_system=autotools libs=shared,static arch=linux-rocky8-x86_64
[^]      ^libdeflate@1.10%gcc@8.5.0 build_system=makefile arch=linux-rocky8-x86_64
[^]          ^gzip@1.12%gcc@8.5.0 build_system=autotools arch=linux-rocky8-x86_64
[^]      ^openssl@1.1.1t%gcc@8.5.0~docs~shared build_system=generic certs=mozilla arch=linux-rocky8-x86_64
[^]          ^ca-certificates-mozilla@2023-01-10%gcc@8.5.0 build_system=generic arch=linux-rocky8-x86_64
[^]      ^zlib@1.2.13%gcc@8.5.0+optimize+pic+shared build_system=makefile arch=linux-rocky8-x86_64
[^]      ^zstd@1.5.5%gcc@8.5.0+programs build_system=makefile compression=none libs=shared,static arch=linux-rocky8-x86_64
```


### Error message

<details><summary>Error message</summary>

<pre>
```console
gavinjohn@n001:/ref/djslab/software/spack-unstable $ rm /r./bin/spack install dorado
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libsigsegv-2.14-luksanijnaeraxwzpglwgez4dpbvoetz
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/berkeley-db-18.1.40-opmuvc3wtmhnsbxo5h3gx5vlx4ddb3qh
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/bzip2-1.0.8-gsgtf2ebbixgbwqdrmlyopopesl4hokl
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/ncurses-6.4-vfg5gfvcjoopc5icro2adk3cxhhk3xsx
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/zlib-1.2.13-fr5epks5vdorwbirha4noqnd2maveiu4
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libiconv-1.17-tetofezsbq26stp5zmif43pezig75moa
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/xz-5.4.1-5imquk52inw2k3n3v2cotqhe2kxaeoug
[+] /ref/djslab/software/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/gcc-runtime-8.5.0-pq3wyldkdxhlvscwppjdbqpxyuqanylv
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libmd-1.0.4-rd5v3y6b5o24fuh4d3g3jaak7q7oqxj5
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/zstd-1.5.5-rsip4fzljuqr3e27dc4wnvkbjeltmys4
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libxcrypt-4.4.33-pxay2kwlr3ynazqphkxr3z6k7rh2okal
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/pcre2-10.42-cxwwjjfvct3qodt7esj7lmbowwytdhpz
[+] /ref/djslab/software/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/gmake-4.4.1-dhtpynyhjt7dvscamkzzjuoino32djt7
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libpciaccess-0.17-apbhxeuuquwvyggz36mo7gd235l5ri2t
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/numactl-2.0.14-lwox3i27kbqfq23djxytc3qyh6kuckcn
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/pkgconf-1.9.5-4dclmaihjknf2epgsncir72ztmfzrzuz
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/gzip-1.12-kouyz4ncka42fvh2wkmrq5qle62vozhv
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/m4-1.4.19-qk62bztedx3a2lntzzqk6lraummgsnk5
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/readline-8.2-qcgehook6czlg5olybo4pel5bntrkeyc
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/openssl-1.1.1t-lvc4tpxxbleehvcr6tfg7li6c7bpsrzn
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/pigz-2.7-bchpnxfkonqojakgxaeumj5gnzzmpzxz
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libunistring-1.1-ypwe625cqsjskxn7a27y32y5r7sldx4b
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libbsd-0.11.7-5k7msasgsi45opxrhwa5olcnviz4gv73
[+] /ref/djslab/software/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libaec-1.0.6-pzwyh4zrt3accbik25gsljmop6ii3isp
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libxml2-2.10.3-ahq6hb4fbezlcsc4cwxp3k7bo2kldod5
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libedit-3.1-20210216-q6m73cxkj4cbdjh75frg4g2x555n2wt7
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libdeflate-1.10-egub4ryg6djdkikfzcgduenj4g25os67
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/gdbm-1.23-vtfsk5dpbhci3nk4ytgddp6kfh3rrj5g
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/curl-8.0.1-r7kfylynz7fq7giyfxbulu3btdtremse
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/cmake-3.26.3-faru6xy5qest4plmddjxgehbdazsv6ki
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libevent-2.1.12-qqtgwwpx4zybjdlxbvo5dieptfvg5jjz
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/tar-1.34-lha7sy5wlcwj4eoo6insr64irgtatlfb
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/libidn2-2.3.4-uzyvgchgysbgwjms3zo4piejd4xgkjm7
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/expat-2.5.0-ch6jqvpttjrwocvy5titnnesu7zjo2ip
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/hwloc-2.9.1-7brlplc6ac4bxic2oj27p3xlr7o7dhwk
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/cuda-11.8.0-muqaot4r2xo3d2osu47ktmupcnvypsyg
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/perl-5.36.0-kupgugdwicduiidrjvvp2x6ly33vrmqk
[+] /ref/djslab/software/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/htslib-1.15.1-nn3oejkikjltlvu34uqq325zjp3pormg
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/gettext-0.21.1-ddq4slysixomsqumk6343adihxyunxyh
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/pmix-4.2.3-m3yrbwysco5ut3l5yzab364r4dochpmo
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/autoconf-2.69-axx6km47pfemzttbxhgts354x2szzqvk
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/krb5-1.20.1-mplv6hlcmukmtqpszay4pkzb5qjgk6aa
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/automake-1.16.5-wshvdxvjitqez2y6tjm3lccfkypqeyal
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/openssh-9.3p1-eqlmsb55uyajqilw5jyvwtn6xpdlx4kp
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/git-2.40.0-o5krgyqcjc3nvzehfyci7scfyhjzmncu
[+] /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/openmpi-4.1.5-xo656bdflf3anqwe3xsijruhp3n4ykdo
[+] /ref/djslab/software/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/hdf5-develop-1.15-n4o36qq7szbr6su7yj7etxu7bb4uh7rs
==> Installing dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih [48/48]
==> No binary for dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih found: installing from source
==> Applied patch /ref/djslab/software/spack-unstable/var/spack/repos/builtin/packages/dorado/cmake-htslib.patch
==> dorado: Executing phase: 'cmake'
==> dorado: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'

4 errors found in build log:
     618    cd /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix/dorado/utils && /ref/djslab/software/spack-unstable/lib/spack/env/gcc/g++ -D
            DORADO_GPU_BUILD=1 -DFMT_BEGIN_NAMESPACE="namespace fmt { inline namespace ont {" -DFMT_END_NAMESPACE=}} -DONLY_C_LOCALE=0 -DSPDLOG_COMPILED_LIB -I/ref/djslab/software/opt/spack/linux-
            rocky8-x86_64/gcc-8.5.0/htslib-1.15.1-nn3oejkikjltlvu34uqq325zjp3pormg/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3
            rdparty/date/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/edlib/edlib/include -I/tmp/gavinjohn/spack-stage/s
            pack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/spdlog/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih
            /spack-src/dorado/3rdparty/ont-minimap2/src/3rdparty/minimap2 -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/to
            rch-2.0.0-ont.2-cxx11-abi-static-Linux/libtorch/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/torch-2.
            0.0-ont.2-cxx11-abi-static-Linux/libtorch/include/torch/csrc/api/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/
            3rdparty/NVTX/c/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/metal-cpp/metal-cpp -isystem /ref/djslab
            /software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/cuda-11.8.0-muqaot4r2xo3d2osu47ktmupcnvypsyg/include -isystem /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x8
            6_64/gcc-8.5.0/cuda-11.8.0-muqaot4r2xo3d2osu47ktmupcnvypsyg/extras/CUPTI/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src
            /dorado/3rdparty/openssl3-Linux-x86_64/include -isystem /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/zlib-1.2.13-fr5epks5vdorwbirha4noqnd2maveiu4/include -
            O3 -DNDEBUG -std=c++17 -Wall -Werror -Wextra -Winvalid-pch -include /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix/dorado/util
            s/pch/CMakeFiles/dorado_pch.dir/cmake_pch.hxx -MD -MT dorado/utils/CMakeFiles/dorado_utils.dir/types.cpp.o -MF CMakeFiles/dorado_utils.dir/types.cpp.o.d -o CMakeFiles/dorado_utils.dir/
            types.cpp.o -c /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/utils/types.cpp
     619    [ 40%] Building CXX object dorado/utils/CMakeFiles/dorado_utils.dir/uuid_utils.cpp.o
     620    cd /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix/dorado/utils && /ref/djslab/software/spack-unstable/lib/spack/env/gcc/g++ -D
            DORADO_GPU_BUILD=1 -DFMT_BEGIN_NAMESPACE="namespace fmt { inline namespace ont {" -DFMT_END_NAMESPACE=}} -DONLY_C_LOCALE=0 -DSPDLOG_COMPILED_LIB -I/ref/djslab/software/opt/spack/linux-
            rocky8-x86_64/gcc-8.5.0/htslib-1.15.1-nn3oejkikjltlvu34uqq325zjp3pormg/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3
            rdparty/date/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/edlib/edlib/include -I/tmp/gavinjohn/spack-stage/s
            pack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/spdlog/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih
            /spack-src/dorado/3rdparty/ont-minimap2/src/3rdparty/minimap2 -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/to
            rch-2.0.0-ont.2-cxx11-abi-static-Linux/libtorch/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/torch-2.
            0.0-ont.2-cxx11-abi-static-Linux/libtorch/include/torch/csrc/api/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/
            3rdparty/NVTX/c/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/metal-cpp/metal-cpp -isystem /ref/djslab
            /software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/cuda-11.8.0-muqaot4r2xo3d2osu47ktmupcnvypsyg/include -isystem /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x8
            6_64/gcc-8.5.0/cuda-11.8.0-muqaot4r2xo3d2osu47ktmupcnvypsyg/extras/CUPTI/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src
            /dorado/3rdparty/openssl3-Linux-x86_64/include -isystem /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/zlib-1.2.13-fr5epks5vdorwbirha4noqnd2maveiu4/include -
            O3 -DNDEBUG -std=c++17 -Wall -Werror -Wextra -Winvalid-pch -include /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix/dorado/util
            s/pch/CMakeFiles/dorado_pch.dir/cmake_pch.hxx -MD -MT dorado/utils/CMakeFiles/dorado_utils.dir/uuid_utils.cpp.o -MF CMakeFiles/dorado_utils.dir/uuid_utils.cpp.o.d -o CMakeFiles/dorado_
            utils.dir/uuid_utils.cpp.o -c /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/utils/uuid_utils.cpp
     621    [ 41%] Building CXX object dorado/utils/CMakeFiles/dorado_utils.dir/cuda_utils.cpp.o
     622    cd /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix/dorado/utils && /ref/djslab/software/spack-unstable/lib/spack/env/gcc/g++ -D
            DORADO_GPU_BUILD=1 -DFMT_BEGIN_NAMESPACE="namespace fmt { inline namespace ont {" -DFMT_END_NAMESPACE=}} -DONLY_C_LOCALE=0 -DSPDLOG_COMPILED_LIB -I/ref/djslab/software/opt/spack/linux-
            rocky8-x86_64/gcc-8.5.0/htslib-1.15.1-nn3oejkikjltlvu34uqq325zjp3pormg/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3
            rdparty/date/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/edlib/edlib/include -I/tmp/gavinjohn/spack-stage/s
            pack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/spdlog/include -I/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih
            /spack-src/dorado/3rdparty/ont-minimap2/src/3rdparty/minimap2 -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/to
            rch-2.0.0-ont.2-cxx11-abi-static-Linux/libtorch/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/torch-2.
            0.0-ont.2-cxx11-abi-static-Linux/libtorch/include/torch/csrc/api/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/
            3rdparty/NVTX/c/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/3rdparty/metal-cpp/metal-cpp -isystem /ref/djslab
            /software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/cuda-11.8.0-muqaot4r2xo3d2osu47ktmupcnvypsyg/include -isystem /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x8
            6_64/gcc-8.5.0/cuda-11.8.0-muqaot4r2xo3d2osu47ktmupcnvypsyg/extras/CUPTI/include -isystem /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src
            /dorado/3rdparty/openssl3-Linux-x86_64/include -isystem /ref/djslab/software/spack-0.20.1/opt/spack/linux-rocky8-x86_64/gcc-8.5.0/zlib-1.2.13-fr5epks5vdorwbirha4noqnd2maveiu4/include -
            O3 -DNDEBUG -std=c++17 -Wall -Werror -Wextra -Winvalid-pch -include /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix/dorado/util
            s/pch/CMakeFiles/dorado_pch.dir/cmake_pch.hxx -MD -MT dorado/utils/CMakeFiles/dorado_utils.dir/cuda_utils.cpp.o -MF CMakeFiles/dorado_utils.dir/cuda_utils.cpp.o.d -o CMakeFiles/dorado_
            utils.dir/cuda_utils.cpp.o -c /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/utils/cuda_utils.cpp
     623    /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/utils/tensor_utils.cpp: In function '_ZN12_GLOBAL__N_123convert_f32_to_f16_implEPN
            3c104HalfEPKfm.resolver':
  >> 624    /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-src/dorado/utils/tensor_utils.cpp:29:43: error: No dispatcher found for f16c
     625     __attribute__((target("avx2,f16c"))) void convert_f32_to_f16_impl(c10::Half* const dest,
     626                                               ^~~~~~~~~~~~~~~~~~~~~~~
  >> 627    make[2]: *** [dorado/utils/CMakeFiles/dorado_utils.dir/build.make:337: dorado/utils/CMakeFiles/dorado_utils.dir/tensor_utils.cpp.o] Error 1
     628    make[2]: *** Waiting for unfinished jobs....
     629    make[2]: Leaving directory '/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix'
  >> 630    make[1]: *** [CMakeFiles/Makefile2:681: dorado/utils/CMakeFiles/dorado_utils.dir/all] Error 2
     631    make[1]: Leaving directory '/tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-bjausix'
  >> 632    make: *** [Makefile:169: all] Error 2

See build log for details:
  /tmp/gavinjohn/spack-stage/spack-stage-dorado-0.5.1-bjausixrkydplpiavm6txe5wsbhgzkih/spack-build-out.txt
```

</pre></details>


### Information on your system

```console
gavinjohn@n001:/ref/djslab/software/spack-unstable $ ./bin/spack debug report
* **Spack:** 0.22.0.dev0 (ecef72c47185eb4ee3e8eed0c5448354c16ac30c)
* **Python:** 3.6.8
* **Platform:** linux-rocky8-cascadelake
* **Concretizer:** clingo
gavinjohn@n001:/ref/djslab/software/spack-unstable $ cat etc/spack/config.yaml 
config:
  install_tree: $spack/../opt/spack
  template_dirs:
    - $spack/../share/spack/templates
  license_dir: $spack/../etc/spack/licenses
  source_cache: $spack/../var/spack/cache
  build_jobs: 128
  ccache: false
gavinjohn@n001:/ref/djslab/software/spack-unstable $ cat etc/spack/upstreams.yaml 
upstreams:
  spack-instance-0-18-1:
    install_tree: /ref/djslab/software/spack-0.18.1/opt/spack
  spack-instance-0-21-0:
    install_tree: /ref/djslab/software/spack-0.21.0/opt/spack
```

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/14670647/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/14670648/spack-build-out.txt)

CC @snehring

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/dorado/package.py
+++ b/var/spack/repos/builtin/packages/dorado/package.py
@@ -16,6 +16,7 @@ class Dorado(CMakePackage, CudaPackage):
 
     maintainers("snehring")
 
+    version("0.5.3", commit="d9af343c0097e0e60503231e036d69e6eda2f19a", submodules=True)
     version("0.5.1", commit="a7fb3e3d4afa7a11cb52422e7eecb1a2cdb7860f", submodules=True)
 
     depends_on("autoconf", type="build")
@@ -29,6 +30,9 @@ class Dorado(CMakePackage, CudaPackage):
     depends_on("zstd")
     depends_on("libdeflate")
     depends_on("zlib-api")
+
+    conflicts("%gcc@:8", msg="Dorado requires at least gcc@9 to compile.")
+    conflicts("%gcc@13:", msg="Dorado will not build with gcc@13 and newer.")
 
     patch("cmake-htslib.patch")
 

